File: conectopia/app/models/Role.scala
package models

import org.hibernate.annotations.{Tuplizer, Target}

import javax.persistence._

@Entity
@Table(name = "role")
class Role extends PersistentEntity {
  // name
  @Column(name = "name")
  var name: String = _
  // color in hex
  @Column(name = "color")
  var color: String = _
  // members one to one
  @OneToOne
  var members: java.util.List[User] = _
  // servers many to one
  @ManyToOne
  @JoinColumn(name = "server_id")
  var server: Server = _
  // chat many to one
  @ManyToOne
  @JoinColumn(name = "chat_id")
  var chat: Chat = _

  // constructor
  def this(name: String, color: String, members: java.util.List[User], server: Server, chat: Chat) {
    this()
    this.name = name
    this.color = color
    this.members = members
    this.server = server
    this.chat = chat
  }
}

File: play-1.5.3/Conectopia/app/models/Role.java
package models;

import javax.persistence.Entity;
import play.db.jpa.Model;

import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Role extends Model {
    public String name;
    public String color;
    @OneToMany(mappedBy = "role")
    public List<Member> members = new ArrayList<>();
    @ManyToOne
    public Server server;
    @ManyToMany
    public List<Chat> chats = new ArrayList<>();

    public Role(String name, String color, Server server) {
        this.name = name;
        this.color = color;
        this.server = server;
    }

    public Role() {
        this.name = "";
        this.color = "";
        this.server = null;
    }
}
